src/write_prometheus_test.c

```diff
@@ -422,18 +397,164 @@ DEF_TEST(target_info) {
   for (size_t i = 0; i < STATIC_ARRAY_SIZE(cases); i++) {
     printf("# Case %zu: %s\n", i, cases[i].name);
 
-    metric_family_t families[cases[i].resources_num];
-    metric_family_t const *family_ptrs[cases[i].resources_num];
+    prometheus_metric_t pms[cases[i].resources_num];
     for (size_t j = 0; j < cases[i].resources_num; j++) {
-      families[j].resource = cases[i].resources[j];
-      family_ptrs[j] = &families[j];
+      pms[j] = (prometheus_metric_t){.resource = cases[i].resources[j]};
+    }
+    prometheus_metric_family_t pfam = {
+        .metrics = pms,
+        .metrics_num = cases[i].resources_num,
+    };
+
+    strbuf_t got = STRBUF_CREATE;
+    target_info(&got, (prometheus_metric_family_t const *[]){&pfam}, 1);
+    EXPECT_EQ_STR(cases[i].want, got.ptr);
+
+    STRBUF_DESTROY(got);
+  }
+
+  return 0;
+}
+
+DEF_TEST(end_to_end) {
+  hostname_set("example.com");
+
+  struct {
+    char const *name;
+    metric_family_t *fams;
+    size_t fams_num;
+    char const *want;
+  } cases[] = {
+      {
+          .name = "single metric",
+          .fams =
+              &(metric_family_t){
+                  .name = "unit.test",
+                  .type = METRIC_TYPE_COUNTER,
+                  .resource =
+                      {
+                          .ptr =
+                              (label_pair_t[]){
+                                  {"host.name", "example.org"},
+                                  {"service.instance.id", "instance1"},
+                                  {"service.name", "name1"},
+                              },
+                          .num = 3,
+                      },
+                  .metric =
+                      {
+                          .ptr =
+                              &(metric_t){
+                                  .value.counter = 42,
+                              },
+                          .num = 1,
+                      },
+              },
+          .fams_num = 1,
+          // clang-format off
+          .want =
+            "# HELP target_info Target metadata\n"
+	    "# TYPE target_info gauge\n"
+	    "target_info{job=\"name1\",instance=\"instance1\",host_name=\"example.org\"} 1\n"
+	    "\n"
+	    "# HELP unit_test_total\n"
+	    "# TYPE unit_test_total counter\n"
+	    "unit_test_total{job=\"name1\",instance=\"instance1\"} 42\n"
+	    "\n"
+	    "# collectd/write_prometheus " PACKAGE_VERSION
+	    " at example.com\n",
```

Comment:
On same line, as in next array item:
```suggestion
	    "# collectd/write_prometheus " PACKAGE_VERSION " at example.com\n",
```

Reply:
✅ done